test/cross-browser/report.js

```diff
@@ -0,0 +1,262 @@
+/*!
+ * test/launch.js
+ */
+
+/* -----------------------------------------------------------------------------
+ * dependencies
+ * ---------------------------------------------------------------------------*/
+
+// Core
+var spawn = require('child_process').spawn,
+    path  = require('path');
+
+/* -----------------------------------------------------------------------------
+ * setup
+ * ---------------------------------------------------------------------------*/
+
+var webdriver = require('selenium-webdriver');
+var By = webdriver.By;
+var logging = webdriver.logging;
+var until = webdriver.until;
+var promise = webdriver.promise;
+
+// hack required to set should on Driver derived promises
+Object.defineProperty(webdriver.promise.Promise.prototype, 'should', {
+    get: Object.prototype.__lookupGetter__('should')
+});
+
+/* -----------------------------------------------------------------------------
+ * reusable
+ * ---------------------------------------------------------------------------*/
+
+var buildDriver = function () {
+    //var prefs = new logging.Preferences();
+    //prefs.setLevel(logging.Type.BROWSER, logging.Level.INFO);
+
+    if (process.env.SAUCE_USERNAME !== undefined) {
+        driver = new webdriver.Builder()
+            .usingServer('http://' + process.env.SAUCE_USERNAME + ':' + process.env.SAUCE_ACCESS_KEY + '@ondemand.saucelabs.com:80/wd/hub')
+            .withCapabilities({
+                //'tunnel-identifier': process.env.TRAVIS_JOB_NUMBER,
+                //build: process.env.TRAVIS_BUILD_NUMBER,
+                username: process.env.SAUCE_USERNAME,
+                accessKey: process.env.SAUCE_ACCESS_KEY,
+                browserName: 'safari'
+            }).build();
+    } else {
+        driver = new webdriver.Builder()
+            .withCapabilities({
+                //browserName: 'chrome'
+                browserName: 'firefox'
+            })
+            //.setLoggingPrefs(prefs)
+            .build();
+    }
+    //.withCapabilities(webdriver.Capabilities.chrome())
+    //.withCapabilities(webdriver.Capabilities.firefox())
+
+    return driver.getWindowHandle();
+};
+
+var wait = function (duration, timeout) {
+    var wait = true;
+    setTimeout(function () { wait = false; }, duration);
+
+    return driver.wait(function () {
+        return !wait;
+    }, timeout || 10000);
+};
+
+var refreshWindow = function () {
+    return driver.navigate().refresh()
+        .then(function () {
+            return driver.executeScript('document.body.style.background = "red";');
+        });
+};
```

Comment:
Do you find this useful? I believe I was previously was utilizing it on `launch.js` so I could visually confirm that the window was refreshing.